easybuild/easyblocks/x/xmipp.py

```diff
@@ -55,112 +60,101 @@ def extract_step(self):
-        # startdir first
-        external_path = os.path.join(self.cfg['start_dir'], 'external')
-        alglib_tar = glob.glob(os.path.join(external_path, 'alglib*.tgz'))[0]
-        for src in ['bilib.tgz', 'bilib.tgz', 'condor.tgz', alglib_tar, 'scons.tgz']:
-            extract_file(os.path.join(external_path, src), external_path)
-
-        # make sure we are back in the start dir
-        os.chdir(self.cfg['start_dir'])
-
-        # build step expects these to exist
-        mkdir(os.path.join(self.cfg['start_dir'], 'bin'))
-        mkdir(os.path.join(self.cfg['start_dir'], 'lib'))
-
-        python_inc_dir = os.path.join(self.python_root, 'include', 'python%s' % self.python_short_ver)
-        numpy_inc_dir = os.path.join(self.python_root, python_libdir, 'numpy', 'core', 'include')
+        """Configure by defining $CONFIGURE_ARGS"""
         if self.toolchain.mpi_family() == toolchain.INTELMPI:
             mpi_bindir = os.path.join(get_software_root('impi'), 'intel64', 'bin')
         else:
             mpi_bindir = os.path.join(get_software_root(self.toolchain.MPI_MODULE_NAME[0]), 'bin')
 
-        if not os.path.exists(numpy_inc_dir):
-            raise EasyBuildError("numpy 'include' directory %s not found", numpy_inc_dir)
-
-        if not os.path.exists(mpi_bindir):
-            raise EasyBuildError("MPI 'bin' subdir %s does not exist", mpi_bindir)
-
-        cmd = ' '.join([
-            self.cfg['preconfigopts'],
-            'python external/scons/scons.py',
-            'mode=configure',
-            '-j %s' % self.cfg['parallel'],
-            '--config=force',
-            'profile=no',
-            'fast=yes',
-            'warn=no',
-            'release=yes',
-            'gtest=no',
-            'cuda=no',
-            'debug=no',
-            'matlab=no',
-            'java=no',
-            'LINKERFORPROGRAMS="$CXX"',
+        root_java = get_software_root("Java")
+        if not get_software_root("Java"):
+            raise EasyBuildError("Module for dependency Java not loaded.")
+
+        configure_args = ' '.join([
+            'profile=no fast=yes warn=no release=yes gtest=yes static=no cuda=no debug=no matlab=no',
+            'LINKERFORPROGRAMS=%s' % os.getenv('CXX'),
             'MPI_BINDIR=%s' % mpi_bindir,
-            'JAVA_HOME=%s' % java_root,
+            'MPI_LIB=mpi',
+            'JAVA_HOME=%s' % os.getenv('JAVA_HOME'),
             'JAVAC=javac',
-            'CC="$CC"',
-            'CXXFLAGS="$CXXFLAGS -DMPICH_IGNORE_CXX_SEEK -I%s -I%s"' % (python_inc_dir, numpy_inc_dir),
-            'CXX="$CXX"',
-            'MPI_CC="$MPICC"',
-            'MPI_CXX="$MPICXX"',
-            'MPI_INCLUDE="$MPI_INC_DIR"',
-            'MPI_LIBDIR="$MPI_LIB_DIR"',
-            'MPI_LINKERFORPROGRAMS="$MPICC"',
-            'LIBPATH="$LD_LIBRARY_PATH"',
-            self.cfg['configopts'],
+            'CC=%s' % os.getenv('CC'),
+            'CXXFLAGS=',
```

Comment:
ok, using Python list syntax works (see http://xmipp.cnb.csic.es/twiki/bin/view/Xmipp/HowToInstall#Windows):

```
'CXXFLAGS=%s' % str(os.getenv('CXXFLAGS').split(' ')).replace(' ', ''),
```

nuts, but it does the job :)